src/controller/index.ts

```diff
@@ -402,6 +407,61 @@ export default class SimulariumController {
         return this.playBackFile;
     }
 
+    public async getMetrics(config: NetConnectionParams): Promise<void> {
+        if (
+            !this.metricsCalculator ||
+            !this.metricsCalculator.socketIsValid()
+        ) {
+            const webSocketClient = new WebsocketClient(config, this.onError);
+            this.metricsCalculator = new RemoteMetricsCalculator(
+                webSocketClient,
+                this.onError
+            );
+            await this.metricsCalculator.connectToRemoteServer();
+        }
+        this.metricsCalculator.getAvailableMetrics();
+    }
+
+    public async getPlotData(
+        config: NetConnectionParams,
+        requestedPlots: PlotConfig[]
+    ): Promise<void> {
+        if (!this.simulator) {
+            return;
+        }
+
+        if (
+            !this.metricsCalculator ||
+            !this.metricsCalculator.socketIsValid()
+        ) {
+            const webSocketClient =
+                this.remoteWebsocketClient &&
+                this.remoteWebsocketClient.socketIsValid()
+                    ? this.remoteWebsocketClient
+                    : new WebsocketClient(config, this.onError);
+            this.metricsCalculator = new RemoteMetricsCalculator(
+                webSocketClient,
+                this.onError
+            );
```

Comment:
Hmm you're definitely right, this does look funky. At this point, I don't remember why I had it set up this way originally, I'll go in and make this more sensible. `getMetrics` will be called at least once before the call to `getPlotData` because we depend on the results from `getMetrics` to know what to request from `getPlotData` (at least we will, once we don't have hard coded values), so that could be related.

- [ ] make sure our connecting to a new WebsocketClient makes sense and is consistent between `getMetrics` and `getPlotData`